### PR TITLE
Initial commit of ROCm-ported ci_build infrastructure

### PR DESCRIPTION
**Pull request by parallelo**

### CODE REVIEW

File: tensorflow/tools/ci_build/target.lst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+gfx803
+gfx900
```
Comment (whchung) [inline tensorflow/tools/ci_build/target.lst]: how about `gfx906`?

@dagamayank  could you weigh in?

**Comment by whchung** [comment]
Merge this PR first. We can add `gfx906` in another PR.

File: tensorflow/tools/ci_build/target.lst (hunk 1 above)
Comment (dagamayank) [reply tensorflow/tools/ci_build/target.lst]: Lets hold off for a bit until we have `gfx906` nodes to use for CI. Any preference to add it right now?

Comment (parallelo) [reply tensorflow/tools/ci_build/target.lst]: I initially added `gfx906`, but noticed quite a few compiler code gen warnings.  When will our compiler release add that support?

**Pull request merged**